libs/market-list/src/lib/market-data-provider.ts

```diff
@@ -0,0 +1,14 @@
+import { makeDerivedDataProvider } from '@vegaprotocol/react-helpers';
+
+import type { Market } from './';
+import { marketsDataProvider } from './';
+
+export const marketDataProvider = makeDerivedDataProvider<Market>(
+  [(callback, client, variables) => marketsDataProvider(callback, client)],
+  ([markets], variables) =>
```

Comment:
chore (non blocking): If find this hard to read. Can we simplify with some nice clear if statements